app/data/Owners.scala

```diff
@@ -0,0 +1,37 @@
+package data
+
+import model._
+
+trait Owners {
+  def stacks: Set[(String, SSA)]
+
+  def all: Set[Owner] = stacks
+    .groupBy(_._1)
+    .map { case (ownerId, stacks) =>
+      Owner(ownerId, stacks.map(_._2))
+    }.toSet
+
+  def find(id: String): Option[Owner] = all.find(_.id == id)
+
+  def find(ssa: SSA): Option[Owner] = all.find(_.hasSSA(ssa))
+
+  def forStack(stackName: Option[String], stageName: Option[String], appName: Option[String]): Option[Owner] = all
+    .find(_.hasSSA(SSA(stackName, stageName, appName)))
+    .orElse(all.find(_.hasSSA(SSA(stackName, stageName))))
+    .orElse(all.find(_.hasSSA(SSA(stackName))))
+    .orElse(all.find(_.hasSSA(SSA())))
+}
+
+object Owners extends Owners {
+
+  override def stacks: Set[(String, SSA)] = Set(
+    "phil.wills" -> SSA(), // Default owner of all stacks
```

Comment:
More like an observation (!), intuitively I think we would search more ofter for the owner of an SSA, rather than the list of SSAs for a specific owner, thus storing the information in a mapping SSA -> Owner where Owner could be only the Owner Id? Just a thought!